libs/design-system/table/src/lib/types/page-table.schema.ts

```diff
@@ -38,20 +38,53 @@ export const MarkdownColumnTypeSchema = z.object({
   type: z.literal('markdown'),
 });
 
+/** Type for Icon Column */
+export type IconColumnType = z.infer<typeof IconColumnTypeSchema>;
+
+/** Schema for Icon Column */
+export const IconColumnTypeSchema = z.object({
+  type: z.literal('icon'),
+  icon: z.string(),
+});
+
 /** Type for Link Column */
 export type LinkColumnType = z.infer<typeof LinkColumnTypeSchema>;
 
-/** Schema for Markdown Column */
+/** Schema for Link Column */
 export const LinkColumnTypeSchema = z.object({
   type: z.literal('link'),
   urlColumn: z.string(),
 });
 
+/** Type for MenuButton Column */
+export type MenuButtonColumnType = z.infer<typeof MenuButtonColumnTypeSchema>;
+
+/** Schema for MenuButton Column */
+export const MenuButtonColumnTypeSchema = z.object({
+  type: z.literal('menu'),
+  icon: z.string(),
+  options: z.string(),
+});
+
+/** Type for MenuOptions Column */
+export type MenuOptionsType = z.infer<typeof MenuOptionsTypeSchema>;
+
+/** Schema for MenuOptions Column */
+export const MenuOptionsTypeSchema = z.object({
```

Comment:
I'm concerned about how these options are structured when loading a csv file. Could you please explain how it will work?